packages/compass-import-export/src/modules/export.ts

```diff
@@ -398,27 +398,17 @@ export const openExport =
   }: {
     namespace: string;
     query: ExportQueryType;
-    // Optional pre supplied count to shortcut and
-    // avoid a possibly expensive re-count.
+    // Optional pre supplied count, this helps us show progress.
     count?: number;
     aggregation?: ExportAggregationType;
-  }): ThunkAction<Promise<void>, RootExportState, void, AnyAction> =>
-  async (dispatch, getState) => {
+  }) =>
+  (dispatch: Dispatch) => {
     const isAggregation = !!aggregation;
     track('Export Opened', { type: isAggregation ? 'aggregation' : 'query' });
-    const {
-      dataService: { dataService },
-    } = getState();
 
     let count = null;
-    try {
-      count =
-        maybeCount ??
-        (!isAggregation
-          ? await fetchDocumentCount(dataService!, namespace, query)
-          : null);
-    } catch (e: unknown) {
-      dispatch(onError(e as Error));
+    if (maybeCount !== undefined) {
+      count = maybeCount;
     }
```

Comment:
At this point the code here basically does `const count = maybeCount ?? null;`, right?